### PR TITLE
PR for the First Milestone

### PR DESCRIPTION
**Pull request by tadeuasarro**
#### Main features:
- Add the Friendship model
- Create relationships between the Friendship and the User models
- Create UI to allow users to send friendship requests, see, accept and reject those requests
- Add test cases for the Friendship, Post, Comment, User, and Like models

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -10,6 +10,6 @@
     Recent posts
   </h3>
   <ul class="posts">
-    <%= render @timeline_posts %>
+    <%= render @friends %>
```
Comment (kubilaycaglayan) [inline app/views/posts/index.html.erb]: :bulb: This line is raising an error. I needed to comment it out to make the application work.
`<%= render @friends %>`

image

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -25,17 +30,21 @@
...
+      <div class="alert-warning p-2 d-flex justify-content-center">
         <p> <%= notice %> </p>
       </div>
     <% end %>
     <% if alert.present? %>
-      <div class="alert">
+      <div class="alert-danger p-2 d-flex justify-content-center">
         <p> <%= alert %> </p>
       </div>
     <% end %>
```
Comment (kubilaycaglayan) [inline app/views/layouts/application.html.erb]: :bulb: Move all the flow control logic out of the views, please 🙏😊

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
### Please ignore this part and scroll down to the last comment that I've dropped.

## :camera_flash: Screenshot
image


### This part ⏬⏬ of the review has done by considering the second milestone, hope it helps you in your next stage. :blush: 

Hello Team 👋
Status: Changes Required ♻️

Great job so far! Thanks for all your effort! 👍👏⭐
You fulfilled most of the requirements 🎉😊
Let's cover a few details to make your project better, together ⏬

### No logic in the views

- [x] There should be absolutely NO logic in views.
    >Views are only for displaying information that became available to it from its corresponding controller not generating it.
**Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

💡 Check below for the inline comments, please ⏬

### Tests
- [x] Add tests for Friendship associations. 🙏

### Posts
- [ ] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

💡 Below ⏬, I marked a line in the code that raises errors when I try to sign-up. I needed to delete that line to proceed and sign-in. Can you please check what is wrong? And make sure that we are able to see the posts.

### That's it, team! After making the changes please request another code review. 💪

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Comment by kubilaycaglayan** [comment]
# :warning: Note: Ignore my review above. And please follow the requirements for the first milestone of this project, that is consist of adding an ERD. Currently, you don't have ERD. 

#### What's happening? 
> You have requested a code review for the first milestone which aligns well with your PR title. But the branch name and all your commits belong to the second milestone. And that confused me at the beginning. Sorry for this inconvenience :blush: 

> Let me know if you think that I am doing something wrong.

> I will not delete my review, since it will be helpful for you on the second milestone.

With all my respect :pray:

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****

_P.S. Make sure to update the live demo link in your README file_ 😄 

 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**